tools/ipapy2/ipapy2_clustering.py

```diff
@@ -1,16 +1,22 @@
-import click
+import argparse
+import sys
 import pandas as pd
 from ipaPy2 import ipa
 
 
-@click.group(invoke_without_command=True)
-@click.option('--i', 'input_filename', type=click.Path(exists=True), required=True)
-@click.option('--o', 'output_filename', type=click.Path(writable=True), required=True)
-def cli(input_filename, output_filename):
-    intensity_table = pd.read_csv(input_filename)
-    result = ipa.clusterFeatures(intensity_table)
-    result.to_csv(output_filename, index=False)
+def main(argv):
+    parser = argparse.ArgumentParser(description="cluster features before IPA pipeline.")
+    parser.add_argument("input_filename", type=str, help="a dataframe containing the measured intensities across several samples.")
+    parser.add_argument("--Cthr", type=float, help="Default value 0.8. Minimum correlation allowed in each cluster.")
+    parser.add_argument("--RTwin", type=float, help="Default value 1. Maximum difference in RT time between features in the same cluster.")
+    parser.add_argument("--Intmode", type=float, help="intensity mode. Default 'max' or 'ave'.")
+    parser.add_argument("output_filename", type=str, help="a dataframe of clustered features.")
```

Comment:
Why is it sometimes `--` prefixed and sometimes not? Also, if there are positional arguments we can't have optional ones in between, as then the position of the positional ones is no longer ensured.